Clean up safe varargs (#33342)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

Previously, these tasks were automatically added to the project for every build.
These tasks are now only added to a project when a rule-based plugin is applied (such as those provided by Gradle's support for building native software).

> ==== Methods on public API types have been made final

Reviewer: ```suggestion
==== Methods on public API types are now `final`
```

> The methods link:{javadocPath}/org/gradle/api/specs/AndSpec.html#findUnsatisfiedSpec(java.lang.Object)[`AndSpec.and`] and link:{javadocPath}/org/gradle/api/reporting/GenerateBuildDashboard.html#aggregate(org.gradle.api.reporting.Reporting...)[`GenerateBuildDashboard.aggregate`] are now final, as required to have the `@SafeVarargs` annotation.

Reviewer: ```suggestion
The methods link:{javadocPath}/org/gradle/api/specs/AndSpec.html#findUnsatisfiedSpec(java.lang.Object)[`AndSpec.and`] and link:{javadocPath}/org/gradle/api/reporting/GenerateBuildDashboard.html#aggregate(org.gradle.api.reporting.Reporting...)[`GenerateBuildDashboard.aggregate`] are now declared as `final`, as required for applying the `@SafeVarargs` annotation.
```

> While these type should not be inherited from, if you build logic or a plugin does it, the build will fail at runtime.

Reviewer: ```suggestion
These types were not intended to be inherited from. 
If your build logic or plugins currently subclass these types and override these methods, you will encounter a runtime failure.
```
